app/javascript/components/supplemental_application/sections/ConfirmedHouseholdIncome.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { isNil } from 'lodash'
+
+import FormGrid from '~/components/molecules/FormGrid'
+import FormGroupTextValue from '~/components/atoms/FormGroupTextValue'
+import { RadioGroup, Radio, Text } from 'react-form';
+
+const YesNoRadioGroup = ({field}) => {
+  return (
+    <div className='radio-group-inline'>
+      <RadioGroup field={field}>
+        {(group) => (
+          <React.Fragment>
+            <p className="radio-inline">
+              <Radio group={group} value={`true`} id={`${field}-yes`}/>
+              <label className="radio-inline_label" htmlFor={`${field}-yes`}>Yes</label>
+            </p>
+            <p className="radio-inline">
+              <Radio group={group} value={`false`} id={`${field}-no`}/>
+              <label className="radio-inline_label" htmlFor={`${field}-no`}>No</label>
+            </p>
+          </React.Fragment>
+        )}
+      </RadioGroup>
+    </div>
+  )
+}
+
+const getAMIPercent = (income, ami) => {
+  console.log('Calculating the ami percent', income)
+  if (isNil(income))
+    return "Enter HH Income"
+  if (isNil(ami))
+    return "Error: AMI unknown"
+  return income / ami
+}
+
+
+
+class ConfirmedHouseholdIncome extends React.Component {
+
+  state = {
+    hudAMI: null,
+    stateAMI: null
+  }
+
+  componentDidMount() {
+    // Display a calculated annual income if applicant only provided monthly income
+    if (isNil(this.props.application.annual_income)) {
+      let annualIncome = this.props.application.monthly_income * 12
```

Comment:
honestly, I'm a bit confused about when to reference props of the component vs formApi.values. should I ever be accessing props?

Reply:
I would say that, `formApi.values` (and setValue) should be used when you are dealing with the form. 
Based on how you are doing it in the this code,  you should use from.values. Values from the form and props might reference different objects and an update on a Text field will not be reflected in the props.